bluemix/src/main/java/com/ibm/watson/SpeechToText.java

```diff
@@ -43,120 +39,157 @@ public String speechToText(byte[] incomingBinaryPOSTData) {
+            if (incomingBinaryPOSTData.length == 0) {
+                return XML_DEC + "<results>" + STATUS_OK + "<transcript />" + "</results>";
+            }
 
             // 'VCAP_SERVICES' contains all the details of services bound to this application
-	    JsonReader jsonReader = Json.createReader(new StringReader(System.getenv("VCAP_SERVICES")));
-	    JsonObject serviceInfo = jsonReader.readObject();
-	    jsonReader.close();
+            JsonObject credentials = readJsonObject(new StringReader(System.getenv("VCAP_SERVICES")))
+                                    .getJsonArray(SERVICE_NAME)
+                                    .getJsonObject(0)
+                                    .getJsonObject("credentials");
 
-            // Get the Service Credentials for SpeechToText
-	    JsonObject credentials = serviceInfo.getJsonArray(SERVICE_NAME)
-		.getJsonObject(0).getJsonObject("credentials");
 
-            try {
-                serviceURL = credentials.getString("url");
-            } catch (Exception e) {}
-            // If we didn't find a URL for the service in VCAP_SERVICES,
-            // use the original.
+
+            serviceURL = credentials.getString("url");
+
             if ("".equals(serviceURL)) {
                 serviceURL = ORIGINAL_SERVICE_URL;
             }
-            
-	    String serviceUser = credentials.getString("username");
-	    String servicePass = credentials.getString("password");
-	    String auth = serviceUser + ":" + servicePass;
-
-	    // Prepare the query string parameters.
-	    // See the Speech to Text service documentation for details.
-	    String queryString = "";
-	    queryString = queryString + "&model=en-US_NarrowbandModel";
-	    queryString = queryString + "&continuous=true";
-
-	    if (DEBUG) {
-		System.out.println("POSTing [" + incomingBinaryPOSTData.length + "] bytes");
-		System.out.println("Using URL [" + serviceURL + API_PATH + "?" + queryString + "]");
-	    }
-
-	    // Prepare the HTTP connection to the service
-	    conn = (HttpURLConnection) new URL(serviceURL + API_PATH + "?" + queryString).openConnection();
-            conn.setDoInput(true);
-            conn.setDoOutput(true);
-            conn.setRequestMethod("POST");
-	    conn.setRequestProperty("Authorization", "Basic " + Base64.encodeBase64String(auth.getBytes()));
-            conn.setRequestProperty("Accept", "*/*");
-            conn.setRequestProperty("Content-Type", "audio/flac");
-            conn.setRequestProperty("Transfer-encoding", "chunked");
-	    conn.setFixedLengthStreamingMode(incomingBinaryPOSTData.length);
-
-	    // Uncomment the following to opt out of improving the service with this audio file
-	    //	    conn.setRequestProperty("X-WDC-PL-OPT-OUT", "1");
-            
+
+
+            // set up our http request
+            conn = setupHttpConnection(incomingBinaryPOSTData, serviceURL, credentials);
```

Comment:
http://stackoverflow.com/questions/40480/is-java-pass-by-reference-or-pass-by-value

TL;DR - everything is pass by value, but you're always passing pointers